src/main/java/io/vertx/rabbitmq/impl/RabbitMQClientImpl.java

```diff
@@ -87,7 +107,11 @@ private static Connection newConnection(RabbitMQOptions config) throws IOExcepti
     cf.setNetworkRecoveryInterval(config.getNetworkRecoveryInterval());
     cf.setAutomaticRecoveryEnabled(config.isAutomaticRecoveryEnabled());
 
-
+    if(config.isSsl()) {   	
+    	SSLHelper sslHelper = new SSLHelper(config, config.getKeyCertOptions(), config.getTrustOptions());
+    	JdkSslContext ctx = (JdkSslContext)sslHelper.getContext((VertxInternal)vertx);
```

Comment:
is this cast required ?

Reply:
Yes it is

Reply:
why ?

Reply:
I see this is a kind of unwrapping, if that is not done what happens ?

Reply:
The RabbitMQ connection factory needs a javax.net.ssl.SSLContext which is wrapped in the JdkSslContext of Netty, So a downcast is required. An OpenSslContext is not supported by the underlying RabbitMQ client.